src/main/java/org/opensrp/web/acl/EventPermissionEvaluator.java

```diff
@@ -0,0 +1,73 @@
+package org.opensrp.web.acl;
+
+import org.opensrp.domain.Event;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.Set;
+
+@Component
+public class EventPermissionEvaluator extends BasePermissionEvaluator<Event> {
+
+	public boolean hasPermission(Authentication authentication, Event targetDomainObject) {
+		return hasPermissionOnEvent(authentication, targetDomainObject.getLocationId());
+	}
+
+	private boolean hasPermissionOnEvent(Authentication authentication, String identifier) {
+		/* @formatter:off */
+		return getAssignedLocations(authentication.getName())
+				.stream()
+				.anyMatch(a -> a.getPlanId().equals(identifier));
+		/* @formatter:on */
+	}
+
+	public boolean hasObjectPermission(Authentication authentication, Serializable targetId, Object permission) {
+		if (targetId instanceof String) {
+			/* @formatter:off */
+			return getAssignedLocations(authentication.getName())
+					.stream()
+					.anyMatch(assignedLocation -> assignedLocation.getPlanId().equals(targetId));
+			/* @formatter:on */
+		} else if (isCollectionOfString(targetId)) {
+			Collection<String> identifiers = (Collection<String>) targetId;
+			/* @formatter:off */
+			return getAssignedLocations(authentication.getName())
+					.stream()
+					.allMatch((assignedLocation) -> {
+						return identifiers.contains(assignedLocation.getPlanId());
+					});
+			/* @formatter:on */
```

Comment:
I don't think it would be possible to check permission for events using Identifiers. Remove this, any where you want to check permission using Identifiers use `PostFilter` or `PostAuthoze` instead